decode.go

```diff
@@ -178,7 +178,9 @@ func (d *decoder) eval(target reflect.Value) {
 				if target.Type() == uuidType {
 					val, err := gouuid.Parse(frame.String)
 					if err != nil {
-						d.error(fmt.Errorf("invalid UUID in Avro encoding: %w", err))
+						if len(frame.String) != 0 {
+							d.error(fmt.Errorf("invalid UUID in Avro encoding: %w", err))
+						} // We allow written "" on encoding zero UUID value, so we do in decoding part
```

Comment:
How about just avoiding parsing entirely when the string is empty?

```
if frame.String == "" {
    // We produce the empty string when encoding the zero UUID value,
    // so allow it when decoding too.
    target.Set(gouuid.UUID{})
    break
}
val, err := gouuid.Parse(frame.String)
etc
```

Reply:
Done in 8213d14, it seems explicit :)